### DISCUSSION TITLE
Support tags

### DISCUSSION THREAD

**Discussion by t-arn** [gsantner/markor#1558] [Q&A] [answered]
I would like to add tags (1 or more) to notes.
It should then be possible to search for notes by tag

**Comment by gsantner** [comment, marked as answer]
This is pretty easy, is interoperable and works in EVERY application & filemanager with file search. Use proper filenames. As you wish, as you like it, up to you.

all my blogposts have -blogpost in it, so I search for that tag "-blogpost". My exports from reddit discussions have reddit in the filename, so I search for reddit.

Plain and simple, not requires vendor specific implementation.

You also have recursive file search with regex support at your hand: `tags: .*mytag.*`

**Reply by t-arn** [reply]
Your response inspired me to another idea: Using html div tags, for example

&lt;div id=tag_work id=tag_job/&gt;

**Reply by aisbergde** [reply]
Plain and simple, I support this. The issue is poor support for long filenames in Markor. If longer filenames would be supported, then tags in filenames work great. I created a separate topic about this.

**Reply by gsantner** [reply]
I thought about allowing multi-line expansion of filenames for a while, but didn't make a decision.

Now made the change, with the next update (v2.8.5 / v2.9.0):
The folder/filenames will then be shown fully, expanded to multiple lines.

**Comment by harshad1** [comment]
@t-arn Note that we have in-file search now. I usually place tags throughout my files and search for them when I need to. Markor is sufficiently fast on my phone and Ripgrep / Ag are fast enough on PC...

**Reply by tomtom** [reply]
Obsidian supports tags in markdown files: #tag1 #tag2 ...

In obsidian, you can click/tap on a tag and you see a list of other occurences. I personally find this super useful and I really would like to see such a feature in markor. The in-file search feature gets you almost there but it is more cumbersome than it needs to be. The "Find in files" feature isn't accessible from the editor view (if I'm not mistaken). So you have to switch back to the files view first.

If somebody considered implementing this in markor, the search for tags would have to be limited to a "project folder" ("vault" in obsidian speech) to be helpful.
